lib/facility.rb

```diff
@@ -1,14 +1,78 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
-
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
+  def initialize(facilities_info)
+    @name = facilities_info[:name]
+    @address = facilities_info[:address]
+    @phone = facilities_info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+  
   end
-
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
-end
+
+  def register_vehicle(vehicle)
+    # return false unless @services.include?('Register Vehicle')
+    vehicle.registration_date = Date.today
+    @registered_vehicles << vehicle
+    vehicle
+  end
+  
+
+  def collected_fees
+    @registered_vehicles.sum { |vehicle| calculated_fees(vehicle) }
+
+  end
+
+
+  def calculated_fees(vehicle)
+    if vehicle.antique?
+      25
+    elsif vehicle.electric?
+      200
+    else
+      100
+    end
+  end
+
+  def administer_written_test(registrant)
+    # return false unless @services.include?('Written Test')
+    if registrant.permit? && registrant.age >= 16
```

Comment:
We can add that 3rd condition here to this line so it would read `if registrant.permit? && registrant.age >= 16 && @services.include?('Written Test')`